examples/how-tos/review-tool-calls.ipynb

```diff
@@ -0,0 +1,762 @@
+    "\n",
+    "There are typically a few different interactions you may want to do here:\n",
+    "\n",
+    "1. Approve the tool call and continue\n",
+    "2. Modify the tool call manually and then continue\n",
+    "3. Give natural language feedback, and then pass that back to the agent instead of continuing\n",
+    "\n",
+    "We can implement this in LangGraph using a [breakpoint](https://langchain-ai.github.io/langgraphjs/how-tos/breakpoints/): breakpoints allow us to interrupt graph execution before a specific step. At this breakpoint, we can manually update the graph state taking one of the three options above\n",
+    "\n",
+    "## Setup\n",
+    "\n",
+    "First we need to install the packages required\n",
+    "\n",
+    "```bash\n",
+    "npm install @langchain/langgraph @langchain/anthropic\n",
+    "```\n",
+    "\n",
+    "Next, we need to set API keys for Anthropic (the LLM we will use)\n",
+    "\n",
+    "```bash\n",
+    "export ANTHROPIC_API_KEY=your-api-key\n",
+    "```\n",
+    "\n",
+    "Optionally, we can set API key for [LangSmith tracing](https://smith.langchain.com/), which will give us best-in-class observability.\n",
+    "\n",
+    "```bash\n",
+    "export LANGCHAIN_TRACING_V2=\"true\"\n",
+    "export LANGCHAIN_CALLBACKS_BACKGROUND=\"true\"\n",
+    "export LANGCHAIN_API_KEY=your-api-key\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Simple Usage\n",
+    "\n",
+    "Let's set up a very simple graph that facilitates this.\n",
+    "First, we will have an LLM call that decides what action to take.\n",
+    "Then we go to a human node. This node actually doesn't do anything - the idea is that we interrupt before this node and then apply any updates to the state.\n",
+    "After that, we check the state and either route back to the LLM or to the correct tool.\n",
+    "\n",
+    "Let's see this in action!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 165,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import { Annotation, MessagesAnnotation, StateGraph, START, END, MemorySaver } from \"@langchain/langgraph\";\n",
+    "import { ChatAnthropic } from \"@langchain/anthropic\";\n",
+    "import { tool } from '@langchain/core/tools';\n",
+    "import { z } from 'zod';\n",
+    "import { ToolMessage, AIMessage, isAIMessage } from '@langchain/core/messages';\n",
+    "\n",
+    "const getWeather = tool((input) => {\n",
+    "    const city = input.city;\n",
+    "    console.log(\"----\");\n",
+    "    console.log(`Searching for: ${city}`);\n",
+    "    console.log(\"----\");\n",
+    "    return \"Sunny!\";\n",
+    "}, {\n",
+    "    name: 'get_weather',\n",
+    "    description: 'Call to get the current weather.',\n",
+    "    schema: z.object({\n",
+    "        city: z.string().describe(\"City to get the weather for.\"),\n",
+    "    })\n",
+    "})\n",
+    "\n",
+    "const State = Annotation.Root({\n",
```

Comment:
Just use `MessagesAnnotation` directly? No need to merge it into an `Annotation.Root`

Reply:
Also - in general our convention is to call these `const StateAnnotation` to avoid `State.State`